src/Command/DrupalUpdate/FileSystemUtility.php

```diff
@@ -0,0 +1,301 @@
+   */
+  protected function removeFile(string $file_path): void {
+    if ($this->fileSystem->exists($file_path)) {
+      $this->fileSystem->remove($file_path);
+    }
+    else {
+      $this->io->note("File " . $file_path . " does not exist.");
+    }
+  }
+
+  /**
+   * @param string $file_url
+   * @param string $save_to
+   * @param string $package
+   * @return false|string|string[]
+   * @throws AcquiaCliException
+   */
+  protected function dumpPackageTarFile(string $file_url, string $save_to, string $package): false|string|array {
+    try {
+      if ($this->downloadFileGuzzleClient($file_url, $save_to . '/' . $package . '.tar.gz')) {
+        return str_replace('.tar.gz', '', basename($file_url));
+      }
+    }
+    catch (Exception $e) {
+      // @todo handle errors
+      throw new AcquiaCliException("Unable to download {$package} file.");
+    }
+    return FALSE;
+  }
+
+  /**
+   * @param string $file_url
+   * @param string $method
+   * @return mixed
+   * @throws AcquiaCliException
+   * @throws GuzzleException
+   */
+  public function getFileContentsGuzzleClient(string $file_url, string $method = 'GET'): mixed {
+    try {
+      $client = $this->getClient();
+      $response = $client->request($method, $file_url);
+
+      if ($response->getStatusCode() !== 200) {
+        return FALSE;
+      }
+      $response = simplexml_load_string($response->getBody()->getContents());
+      $response = json_decode(json_encode($response), TRUE);
+
+      return $response;
+    }
+    catch (Exception $e) {
+      throw new AcquiaCliException("Failed to read {$file_url} .");
+    }
+  }
+
+  /**
+   * @param string $file_url
+   * @param string $save_file_path
+   * @return bool
+   * @throws AcquiaCliException
+   */
+  public function downloadFileGuzzleClient(string $file_url, string $save_file_path): bool {
+    $client = $this->getClient();
+    try {
+      $response = $client->request('GET', $file_url, ['sink' => $save_file_path]);
+      if ($response->getStatusCode() !== 200) {
+        throw new AcquiaCliException("Failed to download {$file_url} .");
+      }
+    }
+    catch (GuzzleException $e) {
+      throw new AcquiaCliException("Failed to download {$file_url} .");
+    }
+    return TRUE;
+  }
+
+  /**
+   * Extract tar.gz files in package path.
+   * @param string $save_to
+   * @param string $package
+   */
+  protected function extractPackage(string $save_to, string $package): void {
+    $phar = new PharData($save_to . '/' . $package . '.tar.gz');
+    $this->fileSystem->remove($save_to . '/' . $package);
+    $phar->extractTo($save_to, NULL, TRUE);
+  }
+
+  /**
+   * @param string $drupal_project_cwd_path
+   * @return array
+   * @throws AcquiaCliException
+   */
+  public function getInfoFilesList(string $drupal_project_cwd_path): array {
+    $finder = new Finder();
+    $finder->files()->in($drupal_project_cwd_path)->name('*.info');
+    $info_package_files = [];
+    if ($finder->count() == 0) {
```

Comment:
We are already throwing exception in command class if count is zero. Lets just return an empty array from here.